contracts/marketplace/src/query.rs

```diff
@@ -221,6 +225,37 @@ pub fn query_ask_renew_price(
     Ok((Some(coin(renewal_price.u128(), NATIVE_DENOM)), valid_bid))
 }
 
+pub fn query_ask_renew_prices(
+    deps: Deps,
+    current_time: Timestamp,
+    token_ids: Vec<String>,
+) -> StdResult<Vec<(Coin, Bid)>> {
+    token_ids
+        .iter()
+        .map(|token_id| {
+            let (coin_option, bid_option) =
+                query_ask_renew_price(deps, current_time, token_id.to_string()).map_err(|e| {
+                    StdError::generic_err(format!(
+                        "Failed to query ask renew price for token_id {}: {}",
+                        token_id, e
+                    ))
+                })?;
+            let coin = coin_option.unwrap_or(Coin {
```

Comment:
I believe these could be optional and omitted instead of having zero values right?

Reply:
yeah for sure

Reply:
resolved in last commits